### PR TITLE
highlight anything that isn't a Basic Latin letter, number, punctuati…

### PR DESCRIPTION
**Pull request by JoshData**
…on, or symbol and replace characters that aren't a letter, number, punctuation, or symbol with their name (since they may not have displayable glyphs)

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,36 +1,62 @@
...
 
-ZWS = flask.Markup('<span class="highlight zero-width-char">zw</span>')
+character_names = { "\n": "\\n", "\r": "\\r", "\t": "\\t" }
 
-def replace_zero_width(text):
-  text = text.replace(u'\u200B', ZWS).replace(u'\u200C', ZWS)
-  return text.replace(u'\u200D', ZWS).replace(u'\uFEFF', ZWS)
+# Load some Unicode data remotely.
+unicode_blocks = []
+for line in urllib.urlopen("http://www.unicode.org/Public/UNIDATA/Blocks.txt").read().split("\n"):
```
Comment (audiodude) [inline app.py]: Yikes, a web request on every request? Why can't we just store a local version of this file? Maybe it gets updated frequently? If that's the case I would prefer to cache the entire file (or even better, a pickle of this function's output) and have a script that updates it when needed.

At the minimum I would say remove the urlopen in this CL and perhaps add the updating logic in a future CL.

File: app.py [hunk 2]
```diff
@@ -1,36 +1,62 @@
...
+  if line.strip() != "" and line[0] != "#":
+    m = re.match("([0-9A-F]+)\.\.([0-9A-F]+); (.*)", line)
+    unicode_blocks.append({
+      "start": int(m.group(1), 16),
+      "end": int(m.group(2), 16),
+      "name": m.group(3),
+    })
+
+def lookup_block(c):
```
Comment (audiodude) [inline app.py]: I would suggest renaming c -> char, it will be more readable.

File: app.py (hunk 1 above)
Comment (JoshData) [inline app.py]: This only executes on app initialization.

**Comment by audiodude** [comment]
Overall this looks really great and promising! The only dealbreaker for me is the urlopen.

File: app.py (hunk 1 above)
Comment (audiodude) [inline app.py]: Good point. I still don't like to depend on an external service like this though. What if it's not available? The app crashes in a mysterious way.

**Comment by JoshData** [comment]
Okay, PR updated with a file hosted in the repo (it's a small file).

File: app.py (hunk 2 above)
Comment (JoshData) [inline app.py]: Fixed.

**Comment by audiodude** [comment]
Thanks for the changes, LGTM.

**Comment by audiodude** [comment]
I've deployed these changes to the website.

**Pull request merged**